docs/reference/commandline/dockerd.md

```diff
@@ -396,61 +396,241 @@ Defaults to 20G.
-      "runtimeArgs": [
-        "--debug"
-      ]
+    "<runtime>": {
+      "runtimeType": "<name-or-path>",
+      "options": {}
+    }
+  }
+}
+```
+
+`runtimeType` refers to either:
+
+- A fully qualified name of a containerd shim.
+
+  The fully qualified name of a shim is the same as `runtime_type` used in
+  CRI containerd configuration. For example, `io.containerd.runsc.v1`.
+
+- The path of a containerd shim binary.
+
+  This option is useful if you installed the containerd shim binary outside of
+  `PATH`.
+
+`options` is optional. It lets you specify the runtime configuration that you
+want to use for the shim. The configuration parameters that you can specify in
+`options` depends on the runtime you're registering. For most shims,
+the supported configuration options are `TypeUrl` and `ConfigPath`.
+For example:
+
+```json
+{
+  "runtimes": {
+    "gvisor": {
+      "runtimeType": "io.containerd.runsc.v1",
+      "options": {
+        "TypeUrl": "io.containerd.runsc.v1.options",
+        "ConfigPath": "/etc/containerd/runsc.toml",
+      }
+    }
+  }
+}
+```
+
+You can configure multiple runtimes using the same runtimeType. For example:
+
+```json
+{
+  "runtimes": {
+    "gvisor-foo": {
+      "runtimeType": "io.containerd.runsc.v1",
+      "options": {
+        "TypeUrl": "io.containerd.runsc.v1.options",
+        "ConfigPath": "/etc/containerd/runsc-foo.toml"
+      }
     },
-    "runc": {
-      "path": "runc"
+    "gvisor-bar": {
+      "runtimeType": "io.containerd.runsc.v1",
+      "options": {
+        "TypeUrl": "io.containerd.runsc.v1.options",
+        "ConfigPath": "/etc/containerd/runsc-bar.toml"
+      }
     }
   }
 }
 ```
 
-This is the same example via the command line:
+If you want to use the runc shims with custom configuration, note that the
+`options` field takes a special set of configuration parameters when used with
+runc shims. For more information about runc parameters, refer to the runc
+configuration section in
+[CRI Plugin Config Guide](https://github.com/containerd/containerd/blob/main/docs/cri/config.md#full-configuration).
+
+###### Configure runc drop-in replacements
+
+If the runtime that you want to register can act as a drop-in replacement for
+runc, use the following format for the runtime entry:
+
+```json
+{
+  "runtimes": {
+    "<runtime>": {
+      "path": "/path/to/binary",
+      "runtimeArgs": ["...args"]
+    }
+  }
+}
+```
+
+Entries with this format use the default containerd runc shim to invoke a
+custom runtime binary, with optional CLI args. For an example, see
+[Alternative container runtimes > youki](https://docs.docker.com/engine/alternative-runtimes/#youki)
+
+##### Register runtimes using the CLI
+
+You can use the `--add-runtime` CLI flag to register runtimes on daemon startup.
+
+You can specify either the name of a fully qualified containerd runtime shim,
+or the name of a registered runtime. The executable must be installed on `PATH`.
```

Comment:
This whole paragraph is incorrect. The `--add-runtime` flag can only be used to register runC drop-in replacements. The flag `--add-runtime myruntime=/path/to/binary` is exactly equivalent to the JSON config
```json
{
  "runtimes": {
    "myruntime": {
      "path": "/path/to/runtime"
    }
  }
}
```

Aside from being unable to define runtime arguments, the `--add-runtime` flag supports the exact same values as the JSON runtime path config. Notably, absolute paths are accepted, and paths which do not contain slashes are looked for in the `PATH` directories, so e.g. `--add-runtime crun=crun` or
```json
{
  "runtimes": {
    "crun": {
      "path": "crun"
    }
  }
}
```
is legal. In fact, the daemon's built-in default runtime configuration is equivalent to `--add-runtime runc=runc`.